Add SAML docs entry for usernameKeys (#105)
sourcegraph/docs: docs/admin/auth/saml/index.mdx

    }
  ```

**usernameAttributeNames**

>   A list of SAML attributes that are used to construct the username for a new user signing up via SAML.

Reviewer: ```suggestion
  A list of SAML attributes to consider for username construction in order or declaration for a new user signing up via SAML.
```

Not sure about my wording here, but somehow wanted to get the ordering in the sentence too :D
